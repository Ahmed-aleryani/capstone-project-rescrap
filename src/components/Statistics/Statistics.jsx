import React from 'react';
import Title from './../Title/Title';
import SubTitle from './../SubTitle/SubTitle';
const statisticsItems = [
  {
    icon: <i class="fas fa-globe-asia"></i>,
    title: '150+',
    subTitle: 'Countries',
  },
  {
    icon: <i class="fas fa-chart-bar"></i>,
    title: '4000+',
    subTitle: 'Registered businesses',
  },
  {
    icon: <i class="fas fa-shopping-cart"></i>,
    title: '2.5m',
    subTitle: 'Pounds of listed material',
  },
];

const Statistics = ({footer}) => {
  return (
    <div>
    <div
      className="container sm:block md:flex justify-between"
      data-testid="statistics"
    >
      {statisticsItems.map((item, index) => {
        return (
          <div className="sm:my-3 flex-1 text-center px-4 py-2 m-2" key={index}>
            <div className="text-green-100 text-6xl">{item.icon}</div>
            <Title classes="" text={item.title} />
            <SubTitle classes="" text={item.subTitle} />
          </div>
        );
      })}
    </div>
    <div className="mt-16">{footer}</div>
    </div>
  );
};

export default Statistics;
