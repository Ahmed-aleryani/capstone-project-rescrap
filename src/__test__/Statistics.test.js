import React from 'react';
import Statistics from './../components/Statistics/Statistics';
import renderer from 'react-test-renderer';

it('matches snapshot', () => {
  const tree = renderer.create(<Statistics />).toJSON();
  expect(tree).toMatchSnapshot();
});
