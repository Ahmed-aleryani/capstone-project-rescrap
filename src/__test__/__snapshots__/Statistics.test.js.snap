// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="container sm:block md:flex justify-between"
    data-testid="statistics"
  >
    <div
      className="sm:my-3 flex-1 text-center px-4 py-2 m-2"
    >
      <div
        className="text-green-100 text-6xl"
      >
        <i
          class="fas fa-globe-asia"
        />
      </div>
      <h1
        className="text-3xl normal"
      >
        150+
      </h1>
      <h2
        className="text-2xl thin"
      >
        Countries
      </h2>
    </div>
    <div
      className="sm:my-3 flex-1 text-center px-4 py-2 m-2"
    >
      <div
        className="text-green-100 text-6xl"
      >
        <i
          class="fas fa-chart-bar"
        />
      </div>
      <h1
        className="text-3xl normal"
      >
        4000+
      </h1>
      <h2
        className="text-2xl thin"
      >
        Registered businesses
      </h2>
    </div>
    <div
      className="sm:my-3 flex-1 text-center px-4 py-2 m-2"
    >
      <div
        className="text-green-100 text-6xl"
      >
        <i
          class="fas fa-shopping-cart"
        />
      </div>
      <h1
        className="text-3xl normal"
      >
        2.5m
      </h1>
      <h2
        className="text-2xl thin"
      >
        Pounds of listed material
      </h2>
    </div>
  </div>
  <div
    className="mt-16"
  />
</div>
`;
